uk.co.compucorp.civicrm.tasksassignments/js/test/services/task_test.js

```diff
@@ -0,0 +1,225 @@
+define([
+  'mocks/data/task',
+  'tasks-assignments/app'
+], function(TaskMock) {
+  'use strict';
+
+  describe('Task', function() {
+    var Task, TaskService, $httpBackend, requestBody, request = {};
+
+    beforeEach(module('civitasks.appDashboard'));
+
+    beforeEach(inject(function(_Task_, _TaskService_, _$httpBackend_) {
+      Task = _Task_;
+      TaskService = _TaskService_;
+      $httpBackend = _$httpBackend_;
+
+      spyOn(Task, 'save').and.callThrough();
+    }));
+
+    beforeEach(function() {
+      $httpBackend.whenGET(/action=getoptions&entity=Task/).respond({});
+      $httpBackend.whenGET(/action=getoptions&entity=Document/).respond({});
+      $httpBackend.whenGET(/action=get&entity=CaseType/).respond({});
+      $httpBackend.whenGET(/action=get&entity=contact/).respond({});
+      $httpBackend.whenGET(/action=get&entity=Task/).respond({});
+      $httpBackend.whenGET(/views.*/).respond({});
+    });
+
+    describe('$resource.save()', function() {
+      describe('when calling TaskService.save()', function () {
+        beforeEach(function() {
+          $httpBackend.whenPOST(/action=create&entity=Task/).respond(function(method, url, data, headers, params) {
+            request.method  = method;
+            request.headers = headers;
+            request.data = data;
+
+            return [200];
+          });
+
+          TaskService.save(TaskMock.task);
+          $httpBackend.flush();
+        });
+
+        it('makes call to save()', function() {
+          expect(Task.save).toHaveBeenCalled();
+        });
+
+        it('makes request with Content-Type as application/x-www-form-urlencoded; charset=UTF-8', function() {
+          expect(request.headers['Content-Type']).toEqual('application/x-www-form-urlencoded; charset=UTF-8');
+        });
+
+        it('adds url encoding to Task list', function () {
+          expect(request.data).toEqual(TaskMock.urlEncodedData.onSave);
```

Comment:
the url encoded data should not be in the mocked data (see my previous comment). You can simply manually create it from the task mock, and then compare it with `request.data`

```js
beforeEach(function () {
  var fakeTask = taskFabricator.single();
  var expectedEncodedData = /* encode the data of fakeTask */
});

it('url encodes the task data', function () {
  expect(request.data).toBe(expectedEncodedData);
});
```